lib/jekyll/converters/scss.rb

```diff
@@ -329,10 +322,21 @@ def sass_generate_source_map(engine)
         Jekyll.logger.warn "Could not generate source map #{e.message} => #{e.cause}"
       end
 
+      # Reads the source-map and adds it to the source-map-page.
       def sass_embedded_generate_source_map(source_map)
         return if associate_page_failed?
 
-        source_map_page.source_map(source_map)
+        map_data = JSON.parse(source_map)
+        map_data["file"] = Addressable::URI.encode(File.basename(output_path))
+        map_data["sources"].map! do |source|
+          if source.start_with? "file:"
+            Addressable::URI.parse(source).route_from(url)
+          else
+            source
+          end
```

Comment:
I feel this could be made compact
```suggestion
          source.start_with?("file:") ? Addressable::URI.parse(source).route_from(url) : source
```